File: kotlin/arc065_a.kt
// ABC049C - 白昼夢

import kotlin.system.exitProcess

fun main() {
  // 文字列を頭から見ていくとdreamerとdreamを見分けにくいため文字列を反転させる
  val FORMAT = listOf("dream", "dreamer", "erase", "eraser").map(String::reversed)
  val S = readLine()!!.reversed()
  var i = 0
  while (i < S.length) {
    var isMatch = false
    FORMAT.forEach {
      if (isMatch || !S.substring(i).startsWith(it)) return@forEach
      isMatch = true
      i += it.length
    }
    if (isMatch) continue
    println("NO")
    exitProcess(0)
  }
  println("YES")
}

File: javascript/arc065_a.js
// ABC049C - 白昼夢

const input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n");
const FORMAT = ["dream", "dreamer", "erase", "eraser"].map(str => str.split("").reverse().join(""));
const S = input[0].split("").reverse().join("");

let i = 0;
while (i < S.length) {
  let isMatch = false;
  FORMAT.forEach(format => {
    if (isMatch || !S.substring(i).startsWith(format)) return; // このreturnはfor文のcontinueと同じ意味
    isMatch = true;
    i += format.length;
  });
  if (isMatch) continue;
  console.log("NO");
  return; // このreturnで実行終了できる
}
console.log("YES");
